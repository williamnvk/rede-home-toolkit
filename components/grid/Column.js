import React from 'react';
import { Column } from './styled';

function renderColumn(props) {
  return <Column {...props} />;
}

export default React.memo(renderColumn);
