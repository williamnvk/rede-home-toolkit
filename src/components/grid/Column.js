// @flow
import React, { memo } from 'react';
import { Component } from './styled';
import { defaultDirection } from '../../constants/flex';
import { ColumnProps } from './types';

function renderColumn(props: ColumnProps) {
  const p = {
    ...props,
    direction: props.direction || defaultDirection,
    size: props.size || 1,
  };
  return <Component {...p} />;
}

export default memo<ColumnProps>(renderColumn);
