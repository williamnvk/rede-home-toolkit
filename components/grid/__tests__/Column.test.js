import React from "react"
import { shallow } from 'enzyme';
import Column from '../Column.js';

describe('Column', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Column />);
    expect(wrapper).toMatchSnapshot();
  });
});
