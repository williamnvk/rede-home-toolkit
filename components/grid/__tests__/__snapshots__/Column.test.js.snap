// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders correctly 1`] = `ShallowWrapper {}`;
